src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Refactoring/RenameEndpoint.cs

```diff
@@ -52,23 +52,24 @@ internal class RenameEndpoint : AbstractRazorDelegatingEndpoint<RenameParams, Wo
             _documentMappingService = documentMappingService ?? throw new ArgumentNullException(nameof(documentMappingService));
         }
 
-        public RegistrationExtensionResult? GetRegistration(VSInternalClientCapabilities clientCapabilities)
+        public RegistrationExtensionResult GetRegistration(VSInternalClientCapabilities clientCapabilities)
         {
             const string ServerCapability = "renameProvider";
             var options = new RenameOptions
             {
                 PrepareProvider = false,
             };
 
-            return new RegistrationExtensionResult(ServerCapability, options);
+            return new RegistrationExtensionResult(ServerCapability, new SumType<bool, RenameOptions>(options));
```

Comment:
❓I'm not very familiar with SumType, why is it not implicitly casted from `RenameOptions`?

Reply:
IIRC the trick here was that when we tried to [set the Property value](https://github.com/dotnet/razor-tooling/blob/features/CLaSP/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ServerCapabilitiesExtensions.cs#L37) for the Registration it threw because the implicit conversion didn't work with `Property.SetValue`.